api/v1beta1/agentserviceconfig_types.go

```diff
@@ -102,6 +102,12 @@ type AgentServiceConfigSpec struct {
 	// +kubebuilder:validation:default:=disabled
 	// +optional
 	IPXEHTTPRoute string `json:"iPXEHTTPRoute,omitempty"`
+	// PublicContainerRegistries is a list of public container registries that will be appended
+	// to the default list (quay.io, registry.svc.ci.openshift.org). Any registry on this list
+	// will not be required to be in the pull secret validated by the assisted-service.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="List of public container registries"
+	// +optional
+	PublicContainerRegistries []string `json:"publicContainerRegistries,omitempty"`
```

Comment:
Do they have to be public?

Reply:
True, they don't. Should this be renamed `ContainerRegistries`?

Reply:
I think that's an improvement. Is there anything else about how they get used and why a user would specify them that might be a useful descriptor in the name?

Reply:
Good point, they're used as part of an ignore list for authenticating a user's pull secret. 

What do you think about these options?

1. `IgnoredAuthContainerRegistries`
2. `PullSecretIgnoreRegistries`
3. `NoAuthContainerRegistries`

Reply:
`NoAuthContainerRegistries` sounds intuitive.

[Searching the openshift api repo](https://github.com/openshift/api/search?q=registry), it looks like the trend is to call them a "registry" or "image registry", but not a "container registry". I think that makes sense since the registry is full of images that can be used to create a container (which is a runtime concept).

How about something like this?

```
// NoAuthRegistries is a list of registries from which container images can be pulled
// without authentication. They will be appended to the default list (quay.io,
// registry.svc.ci.openshift.org). Any registry on this list will not be required to be in
// the pull secret validated by the assisted-service.
```

If there's a little more info about how/where this gets used, that would probably be useful. Is it just for the images that get accessed by the discovery ISO?

Reply:
Those both sound good! Thanks for looking up the convention for calling it just "registry" or "image registry". Maybe Nick's suggestion sounds more seamless? `UnauthenticatedRegistries`

> Is it just for the images that get accessed by the discovery ISO?

It looks like the pull secret is also used by the openshift installer so I think it would be used to pull the RHCOS image and other OCP operator images (such as MCO)

Reply:
Updated the PR to use `UnauthenticatedRegistries`! How does it look?